Lzhl_tcp.cpp

```diff
@@ -212,7 +223,19 @@ int lzhl_recv( SOCKET sock, char* buf, int bufSz, int flags )
 			ls.dSz = dataSz;
 			ls.dDisp = 0;
 			ls.dBuf = new BYTE[ dataSz ];
+
+#if _WIN32
 			int Ok = LZHLDecompress( dh, ls.dBuf, &dataSz, compBuf, &compSz );
+#else
+            size_t dataSzLongUint = (size_t)dataSz;
+            size_t compSzLongUint = (size_t)compSz;
+            int Ok = LZHLDecompress( dh, ls.dBuf, &dataSzLongUint, compBuf, &compSzLongUint );
+            assert( dataSzLongUint <= UINT_MAX );
+            assert( compSzLongUint <= UINT_MAX );
+            dataSz = (unsigned int)dataSzLongUint;
+            compSz = (unsigned int)compSzLongUint;
```

Comment:
It looks like `dataSz`, `compSz` are not used after this point so you can skip the assignment and assert here.